### ISSUE TITLE
Eirini enabled KubeCF deployment failure

### ISSUE THREAD

**Issue by lordcf** [cloudfoundry-incubator/kubecf#1057]
**Eirini enabled KubeCF deployment failure**
Case 1:
Tried KubeCF deployment on KIND cluster with eirini enabled feature, all the pods goes into running state except the "bits" pod. The bits pod remains in "ContainerCreating" state. Tried cf login and pushing the sample app into CF. The apps shows "unable to start the app" state and deployment of the app fails. 

![image](https://user-images.githubusercontent.com/25821409/85382197-5374be80-b55c-11ea-95a4-6fdabb65c0bd.png)

Below are the events for the sample deployed app:
Events:
  Type     Reason       Age                   From                     Message
  ----     ------       ----                  ----                     -------
  Warning  FailedMount  27m (x9 over 66m)     kubelet, pallavi-worker  Unable to attach or mount volumes: unmounted volumes=[bits-cert], unattached volumes=[bits-config bits-cert default-token-qp4gk]: timed out waiting for the condition
  Warning  FailedMount  23m (x8 over 61m)     kubelet, pallavi-worker  Unable to attach or mount volumes: unmounted volumes=[bits-cert], unattached volumes=[default-token-qp4gk bits-config bits-cert]: timed out waiting for the condition
  Warning  FailedMount  7m17s (x4 over 54m)   kubelet, pallavi-worker  Unable to attach or mount volumes: unmounted volumes=[bits-cert], unattached volumes=[bits-cert default-token-qp4gk bits-config]: timed out waiting for the condition
  Warning  FailedMount  2m54s (x40 over 68m)  kubelet, pallavi-worker  MountVolume.SetUp failed for volume "bits-cert" : secret "bits-service-ssl" not found

Case 2:
 Tried eirini enabled KubeCF deployment on KIND cluster with modifying "useExistingSecret: true" in values.yaml. Getting below error while running the install KubeCF command:

Error: template: kubecf/charts/bits/templates/certificate.yaml:3:17: executing "kubecf/charts/bits/templates/certificate.yaml" at <fail "secrets.BITS_TLS_CRT must be set">: error calling fail: secrets.BITS_TLS_CRT must be set
 
** KubeCF installation steps followed  **
1) Created the KIND cluster 
2) Installed cf-operator v4.5.6 (helm v3.2)
3) Installed Kubecf v2.2.2
( Command : helm install kubecf --namespace kubecf --values deploy/helm/kubecf/values.yaml https://kubecf.s3.amazonaws.com/kubecf-v2.2.2.tgz )

**Expected behavior**
The KubeCF deployment must be successful and can deploy apps successfully with eirini enabled feature.

**Comment by jimmykarily** [comment]
Hi @lordcf . Let's ignore "Case 2" for a moment (you are getting an error because you try to create an existing secret when such a secret is not defined). For case one, it seems as if this is not created: https://github.com/cloudfoundry-incubator/kubecf/blob/master/deploy/helm/kubecf/templates/eirini.yaml#L11

can you please try these and send us what you see?

1. `kubectl get quarkssecrets -n kubecf` (there should be a `bits-service-ssl` one)
2 .`kubectl get pvc -n kubecf` (no volumes should be "pending" or anything)
3. Check the quarks-job pod logs for relevant errors. There will be lots of output, try to find errors or things that look relevant. The quarkssecret is converted to a kubernetes "Secret" and if that is failing you should be able to see an error in that pod.

**Comment by lordcf** [comment]
Hello @jimmykarily ,
Below is the output of the commands suggested by you:
1. kubectl get quarkssecrets -n kubecf
- > bits-service-ssl is visible in the list
output:
$ kubectl get quarkssecrets -n kubecf
NAME                                                      AGE
**bits-service-ssl                                          3h29m**
var-adapter-rlp-tls                                       3h29m
var-adapter-tls                                           3h29m
var-application-ca                                        3h29m
var-blobstore-admin-users-password                        3h29m
.....

2. kubectl get pvc -n kubecf
- > No Volumes are in pending state
**output**: 
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                                                  STORAGECLASS   REASON   AGE
pvc-4b498660-c319-494d-851a-8ab95c364315   100Gi      RWO            Delete           Bound    kubecf/singleton-blobstore-pvc-singleton-blobstore-0   local-path              44m
pvc-9b949d8a-3063-4e7f-9a19-75d640f8d4f7   100Gi      RWO            Delete           Bound    kubecf/pxc-data-database-0                             local-path              48m

3. Checked the quarks-job pod logs. No quarkssecret related error we found in "cf-operator-quarks-job" pod, also no other error found in the logs.

**Comment by jimmykarily** [comment]
You have the quarks secret which means the corresponding secret should be generated. I'm thinking that:
- the secret is generated in the wrong namespace (check which secrets you have in all namespaces to see if `bits-service-ssl` exists somewhere).
- The secret is not generated for some reason. In this case there must be an error message somewhere in quarks or quarks-job pod. @manno any suggestion on what kind of text to look for?

**Comment by lordcf** [comment]
hi @jimmykarily , checked for the quarks secret in all namespaces but couldn't find it. Can you please help what command or text we need to search for?

**Comment by viovanov** [comment]
Could be the same as #989

**Comment by danielvaughan** [comment]
I am having the same problem in both 2.2.2 and 2.3.0 on k8s v1.17.6 (rke)

I have the quarks secret
`kubectl get quarkssecrets bits-service-ssl      
NAME               AGE
bits-service-ssl   75m
`
my pvcs are ok
`k get pvc
NAME                                            STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
pxc-data-database-0                             Bound    pvc-10583be2-a5aa-45a8-8b31-e51806cc0078   20Gi       RWO            longhorn       74m
singleton-blobstore-pvc-singleton-blobstore-0   Bound    pvc-329eb5a7-9fde-48ab-a297-4186b20e3e53   100Gi      RWO            longhorn       73m`

but I cannot see a bits-service-ssl   secret

`k get secret --all-namespaces | grep bits
kubecf              bits                                                                                         Opaque                                1      77m
kubecf              bits-service-token-zp78d                                                                     kubernetes.io/service-account-token   3      77m
kubecf              kubecf-bits-service-ssl-csr-private-key                                                      Opaque                                2      77m
`
As a result my bits is
```
bits-6f8b84ff98-bsnh6                      0/1     ContainerCreating   0          29m
```

with the events

```
Events:
  Type     Reason       Age                   From               Message
  ----     ------       ----                  ----               -------
  Normal   Scheduled    29m                   default-scheduler  Successfully assigned kubecf/bits-6f8b84ff98-bsnh6 to x230a
  Warning  FailedMount  18m                   kubelet, x230a     Unable to attach or mount volumes: unmounted volumes=[bits-cert], unattached volumes=[bits-cert default-token-p5w9m bits-config]: timed out waiting for the condition
  Warning  FailedMount  14m (x2 over 27m)     kubelet, x230a     Unable to attach or mount volumes: unmounted volumes=[bits-cert], unattached volumes=[default-token-p5w9m bits-config bits-cert]: timed out waiting for the condition
  Warning  FailedMount  7m23s (x7 over 25m)   kubelet, x230a     Unable to attach or mount volumes: unmounted volumes=[bits-cert], unattached volumes=[bits-config bits-cert default-token-p5w9m]: timed out waiting for the condition
  Warning  FailedMount  3m14s (x21 over 29m)  kubelet, x230a     MountVolume.SetUp failed for volume "bits-cert" : secret "bits-service-ssl" not found

```

**Comment by danielvaughan** [comment]
one 2.3.0 cluster working now and one not. The difference in the quarks-operator logs is the working one does

```
2020-07-16T15:50:36.073Z	DEBUG	csr-reconciler	quarkssecret/certificatesigningrequest_reconciler.go:181	Waiting for CSR to be issued: CSR kubecf-bits-service-ssl has already been approved
2020-07-16T15:50:36.073Z	DEBUG	controller-runtime.controller	controller/controller.go:282	Successfully Reconciled	{"controller": "certificate-signing-request-controller", "request": "/kubecf-bits-service-ssl"}
2020-07-16T15:50:36.084Z	INFO	csr-reconciler	quarkssecret/certificatesigningrequest_reconciler.go:57	Reconciling CSR 'kubecf-bits-service-ssl'
2020-07-16T15:50:36.084Z	DEBUG	csr-reconciler	quarkssecret/certificatesigningrequest_reconciler.go:72	CSR 'kubecf-bits-service-ssl' is in 'issued' state
2020-07-16T15:50:36.085Z	INFO	csr-reconciler	quarkssecret/certificatesigningrequest_reconciler.go:142	Creating certificate secret 'bits-service-ssl' for CSR 'kubecf-bits-service-ssl'
2020-07-16T15:50:36.085Z	DEBUG	csr-reconciler	quarkssecret/certificatesigningrequest_reconciler.go:204	Creating secret 'bits-service-ssl'
2020-07-16T15:50:36.110Z	DEBUG	csr-reconciler	quarkssecret/certificatesigningrequest_reconciler.go:215	Secret 'bits-service-ssl' has been created
2020-07-16T15:50:36.110Z	DEBUG	csr-reconciler	quarkssecret/certificatesigningrequest_reconciler.go:221	Deleting secret 'kubecf-bits-service-ssl-csr-private-key'
```

The failing one does not gets to 
```
2020-07-16T14:35:37.773Z        DEBUG   csr-reconciler  quarkssecret/certificatesigningrequest_reconciler.go:197        CSR 'kubecf-bits-service-ssl' has been updated
```

but does not do the next step. No errors in the log

**Comment by viovanov** [comment]
Sounds like this is fixed in 2.4.0 based on feedback from @jbuns
